File: moviper/src/main/java/com/mateuszkoslacz/moviper/base/exception/PresenterInstancesAccessNotEnabled.java
package com.mateuszkoslacz.moviper.base.exception;

import android.app.Application;

import com.mateuszkoslacz.moviper.presenterbus.Config;
import com.mateuszkoslacz.moviper.presenterbus.Moviper;

/**
 * Created by mateuszkoslacz on 24.10.2016.
 * <p>
 * Thrown when you haven't enabled IPC Presenter Instance Access and you try to use it ({@link
 * Moviper#getPresenterInstance(Class, String)}. To avoid that configure Moviper using {@link
 * com.mateuszkoslacz.moviper.presenterbus.Moviper#setConfig(Config)} in {@link
 * Application#onCreate()} with {@link Config.Builder#withInstancePresentersEnabled(boolean)} set to
 * true.
 */


public class PresenterInstancesAccessNotEnabled extends RuntimeException {

    public PresenterInstancesAccessNotEnabled() {
        super("You can't call getPresenterInstance() without explicitly enabling presenter " +
                "instances access! Instantiate Moviper in Application class using " +
                "Moviper.getInstance().setConfig(new Config.Builder()...) and setting " +
                "withInstancePresentersEnabled(true) in config builder.");
    }
}

File: moviper3/src/main/java/com/mateuszkoslacz/moviper3/exception/PresenterInstancesAccessNotEnabled.kt
package com.mateuszkoslacz.moviper3.exception

import android.app.Application

import com.mateuszkoslacz.moviper3.presenterbus.Config
import com.mateuszkoslacz.moviper3.presenterbus.Moviper

/**
 * Created by mateuszkoslacz on 24.10.2016.
 *
 *
 * Thrown when you haven't enabled IPC Presenter Instance Access and you try to use it ([ ][Moviper.getPresenterInstance]. To avoid that configure Moviper using [ ][Moviper.setConfig] in [ ][Application.onCreate] with [Config.Builder.withInstancePresentersEnabled] set to
 * true.
 */
class PresenterInstancesAccessNotEnabled
    : RuntimeException("You can't call getPresenterInstance() without explicitly enabling presenter " +
        "instances access! Instantiate Moviper in Application class using " +
        "Moviper.getInstance().setConfig(new Config.Builder()...) and setting " +
        "withInstancePresentersEnabled(true) in config builder.")
